src/Components/App/App.js

```diff
@@ -0,0 +1,59 @@
+import './App.css'
+import React, { Component } from 'react'
+import { Route, Switch } from 'react-router-dom'
+import { getHikes } from '../../apiCalls'
+import Header from '../Header/Header'
+import HikeDetails from '../HikeDetails/HikeDetails'
+import FavoriteHikes from '../FavoriteHikes/FavoriteHikes'
+import Hikes from '../Hikes/Hikes'
+
+class App extends Component {
+  constructor() {
+    super()
+    this.state = {
+      hikes: [],
+      favoriteHikes: [],
+      error: ''
+    }
+  }
+
+  componentDidMount = async () => {
+    try {
+      const result = await getHikes()
+      const data = await result.json()
+
+      this.setState({hikes: data.hikes})
+    } catch (error) {
+      this.setState({
+        error: 'There was an error fetching the hikes'
+      })
+    }
+  }
+
+  saveFavoriteHike = (id) => {
+    const favHike = this.state.hikes.find(hike => hike.id === id)
+    this.state.favoriteHikes.push(favHike)
```

Comment:
for this part, you don't want to change the state directly. 

you would want to either make a copy of the state and update it that way let me know how I can help